### PR TITLE
Db users

### PR DESCRIPTION
**Pull request by alancuenca**

### CODE REVIEW

File: db/models/users.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            INSERT INTO users (username, email, password)
+            VALUES ($1, $2, $3)
+            RETURNING *;
+        `, [username, email, hashPwd]);
+        delete user.password
+        return user;
+    } catch (error) {
+        throw error;
+    }
+}
```
Comment (scburkha005) [inline db/models/users.js:18]: I'd recommend thinking about what guard clauses we can add to these statements. For example, if the username OR the password are undefined, we can throw a custom error to address that.
`throw {
  name: "FieldsRequired"
  message: "You must provide both a username and password"
}`

these kinds of guard clauses can be helpful once we go to building our front end by giving us very clear and detailed error messages for what went wrong.

**Review by scburkha005 — CHANGES_REQUESTED** [review]
Code looks good, we may need additional db-adapter functions but this is a good basis to start with.
examples:
- what if the user wants to update their email?
- do we let users delete their account?
we may not need to include these for now, but if we want to add more features in the future we may come back and add these in.

File: db/models/users.js (hunk 1 above)
Comment (alancuenca) [reply db/models/users.js:18]: You got it. I'll work on that now.

**Review by scburkha005 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**